### PR TITLE
Updates docker_registry2 from 1.18.0 to 1.18.2 to fix quay.io container registry pagination

### LINKED ISSUE #10347: Dependabot not following pagination on quay.io, again
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

docker

### Package manager version

_No response_

### Language version

_No response_

### Manifest location and content before the Dependabot update

`/Dockerfile`

```dockerfile
FROM quay.io/keycloak/keycloak:24.0.5
...
```

### dependabot.yml content

```yaml
version: 2

updates:
  - package-ecosystem: "docker"
    directory: "/"
    schedule:
      interval: "daily"
```

### Updated dependency

`quay.io/keycloak/keycloak:24.0.5`

### What you expected to see, versus what you actually saw

- expected: `quay.io/keycloak/keycloak:25.0.2` (bumped)
- actual: `quay.io/keycloak/keycloak:24.0.5` (not bumped)

### Native package manager behavior

_No response_

### Images of the diff or a link to the PR, issue, or logs

```
2024/08/01 14:06:47 INFO <job_863701123> Checking all dependencies for version updates...
updater | 2024/08/01 14:06:47 INFO <job_863701123> Checking if keycloak/keycloak 24.0.5 needs updating
  proxy | 2024/08/01 14:06:47 [012] GET https://quay.io:443/v2/keycloak/keycloak/tags/list
  proxy | 2024/08/01 14:06:47 [012] 200 https://quay.io:443/v2/keycloak/keycloak/tags/list
updater | 2024/08/01 14:06:47 INFO <job_863701123> Latest version is 24.0.5
updater | 2024/08/01 14:06:47 INFO <job_863701123> No update needed for keycloak/keycloak 24.0.5
```

### Smallest manifest that reproduces the issue

_No response_

### PR DESCRIPTION
**Pull request by thomas-riccardi**
### What are you trying to accomplish?

what: Fix #10347: fix quay.io container registry pagination, cf https://github.com/deitch/docker_registry2/issues/106

how: Updates `docker_registry2` from 1.18.0 to 1.18.2 (I copied the diff from #10480)

### Anything you want to highlight for special attention from reviewers?

Automatic dependabot PRs don't seem to be merged, so I try a manual PR.

Updates `docker_registry2` from 1.18.0 to 1.18.2

Commits
* [`b830597`](https://github.com/deitch/docker_registry2/commit/b83059790180fe909ff73471a4e810f9afee7ddd) bump 1.18.2 ([#108](https://redirect.github.com/deitch/docker_registry2/issues/108))
* [`2915d80`](https://github.com/deitch/docker_registry2/commit/2915d803cac7a3c2efa0f8452502ee9d806da221) Fix pagination for tags list pagination for quay.io & registry.access.redhat....
* [`2071379`](https://github.com/deitch/docker_registry2/commit/2071379d3dd6bf0647ac3c54435620d5bc8c22c8) remove gemfile.lock from gitignore and commit to git ([#105](https://redirect.github.com/deitch/docker_registry2/issues/105))
* [`c33e9f5`](https://github.com/deitch/docker_registry2/commit/c33e9f5a33669922b157b46fd6aec4980e5a0338) v1.18.1 ([#104](https://redirect.github.com/deitch/docker_registry2/issues/104))
* [`4d4b968`](https://github.com/deitch/docker_registry2/commit/4d4b968dfdf20bd9a437a2afa400ff3a0a218df4) Smartly join URLs ([#101](https://redirect.github.com/deitch/docker_registry2/issues/101))
* [`d11771a`](https://github.com/deitch/docker_registry2/commit/d11771adf6d2371cf43663609656237b170dc301) Enhancement of rubocop and tests configs ([#103](https://redirect.github.com/deitch/docker_registry2/issues/103))
* [`4c130c9`](https://github.com/deitch/docker_registry2/commit/4c130c928d0df3e976993f9a6991717a9d6a1a79) Fix rubocop on master ([#102](https://redirect.github.com/deitch/docker_registry2/issues/102))
* See full diff in [compare view](https://github.com/deitch/docker_registry2/compare/v1.18.0...v1.18.2)

### How will you know you've accomplished your goal?

cf issue #10347

### Checklist

- [ ] I have run the complete test suite to ensure all tests and linters pass.
- [ ] I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
- [x] I have written clear and descriptive commit messages.
- [x] I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
- [x] I have ensured that the code is well-documented and easy to understand.

### CODE REVIEW

File: updater/Gemfile.lock [hunk 1]
```diff
@@ -174,7 +174,7 @@ GEM
       reline (>= 0.3.8)
     diff-lcs (1.5.1)
     docile (1.4.0)
-    docker_registry2 (1.18.0)
```
Comment (yeikel) [inline updater/Gemfile.lock:177]: Although this https://github.com/dependabot/dependabot-core/blob/f238da5b58dabcf59014f8f6ec2409f6ea3804b6/common/dependabot-common.gemspec#L33 definition covers this change, we should consider upgrading there instead

Comment (thomas-riccardi) [reply updater/Gemfile.lock:177]: ok, I have pushed commits with the modification in `dependabot-core/common/dependabot-common.gemspec`, and also manually in `Gemfile.lock` and `updater/Gemfile.lock`.
Unsure about all that, notably about `Gemfile.lock`.
I'm not a ruby developer, I'm not familiar with the ruby packaging, but `docker_registry2` 1.18.0..1.18.2 has no dependency change (except dev dependency), so it should be ok to patch the Gemfile.lock manually.

**Comment by thomas-riccardi** [comment]
Sorbet check fails, I cannot reproduce the correct environment, to run the suggested `bin/tapioca gem`: `bundle exec tapioca gem --verify`
```
$ bin/docker-dev-shell bundler
 > image dependabot/dependabot-core-development-bundler already exists

=> running docker development shell
[dependabot-core-dev] ~ $ cd dependabot-updater/
[dependabot-core-dev] ~/dependabot-updater $ bundle exec tapioca gem --verify
bundler: failed to load command: tapioca (/home/dependabot/bin/tapioca)
/usr/local/lib/ruby/site_ruby/3.3.0/bundler/rubygems_integration.rb:265:in `block in replace_bin_path': can't find executable tapioca for gem tapioca. tapioca is not currently included in the bundle, perhaps you meant to add it to your Gemfile? (Gem::Exception)
```

**Comment by targylyen** [comment]
@thomas-riccardi should that work against branch `main` as well ? cause this is what happening:

```
dependabot-core % git branch
* main
dependabot-core % ./bin/docker-dev-shell bundler
 > image dependabot/dependabot-core-development-bundler already exists

=> running docker development shell
[dependabot-core-dev] ~ $ cd dependabot-updater/
[dependabot-core-dev] ~/dependabot-updater $ bundle exec tapioca gem --verify
bundler: failed to load command: tapioca (/home/dependabot/bin/tapioca)
/usr/local/lib/ruby/site_ruby/3.3.0/bundler/rubygems_integration.rb:265:in `block in replace_bin_path': can't find executable tapioca for gem tapioca. tapioca is not currently included in the bundle, perhaps you meant to add it to your Gemfile? (Gem::Exception)
        from /usr/local/lib/ruby/site_ruby/3.3.0/bundler/rubygems_integration.rb:304:in `block in replace_bin_path'
        from /home/dependabot/bin/tapioca:6:in `<top (required)>'
        from /usr/local/lib/ruby/site_ruby/3.3.0/bundler/cli/exec.rb:58:in `load'
        from /usr/local/lib/ruby/site_ruby/3.3.0/bundler/cli/exec.rb:58:in `kernel_load'
        from /usr/local/lib/ruby/site_ruby/3.3.0/bundler/cli/exec.rb:23:in `run'
        from /usr/local/lib/ruby/site_ruby/3.3.0/bundler/cli.rb:455:in `exec'
        from /usr/local/lib/ruby/site_ruby/3.3.0/bundler/vendor/thor/lib/thor/command.rb:28:in `run'
        from /usr/local/lib/ruby/site_ruby/3.3.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
        from /usr/local/lib/ruby/site_ruby/3.3.0/bundler/vendor/thor/lib/thor.rb:527:in `dispatch'
        from /usr/local/lib/ruby/site_ruby/3.3.0/bundler/cli.rb:35:in `dispatch'
        from /usr/local/lib/ruby/site_ruby/3.3.0/bundler/vendor/thor/lib/thor/base.rb:584:in `start'
        from /usr/local/lib/ruby/site_ruby/3.3.0/bundler/cli.rb:29:in `start'
        from /usr/local/lib/ruby/gems/3.3.0/gems/bundler-2.5.16/exe/bundle:28:in `block in <top (required)>'
        from /usr/local/lib/ruby/site_ruby/3.3.0/bundler/friendly_errors.rb:117:in `with_friendly_errors'
        from /usr/local/lib/ruby/gems/3.3.0/gems/bundler-2.5.16/exe/bundle:20:in `<top (required)>'
        from /usr/local/bin/bundle:25:in `load'
        from /usr/local/bin/bundle:25:in `<main>'
```

**Comment by thomas-riccardi** [comment]
@targylyen I have no idea, I'm just trying to bump a dependency on dependabot-core to get a fix, I'm not a ruby developer.

**Comment by lewis-strong** [comment]
> @targylyen I have no idea, I'm just trying to bump a dependency on dependabot-core to get a fix, I'm not a ruby developer.

I've created a PR on your fork to hopefully update this to sort the pipeline sorbet failure

**Comment by lewis-strong** [comment]
> > @targylyen I have no idea, I'm just trying to bump a dependency on dependabot-core to get a fix, I'm not a ruby developer.
> 
> I've created a PR on your fork to hopefully update this to sort the pipeline sorbet failure

Once you've done that you should rebase this PR too

**Comment by targylyen** [comment]
> > > @targylyen I have no idea, I'm just trying to bump a dependency on dependabot-core to get a fix, I'm not a ruby developer.
> > 
> > 
> > I've created a PR on your fork to hopefully update this to sort the pipeline sorbet failure
> 
> Once you've done that you should rebase this PR too

@thomas-riccardi are you able to take a look at this ?

**Comment by lewis-strong** [comment]
Alternatively I've opened https://github.com/dependabot/dependabot-core/pull/10855 which includes these changes and the Sorbet tests are now passing

**Comment by thomas-riccardi** [comment]
@lewis-strong 
I'm fine with closing here and just use you PR #10855: I don't know how to review your commit.
But if you prefer, I can rebase here to latest master

**Comment by lewis-strong** [comment]
> @lewis-strong 
> I'm fine with closing here and just use you PR #10855: I don't know how to review your commit.
> But if you prefer, I can rebase here to latest master
>That's fine and we can use the PR I raised which has all the checks passed to speed things up 👍🏻 No need to rebase your branch if you're going to close this PR

**Comment by thomas-riccardi** [comment]
closing in favor of #10855

**Pull request closed without merging**